src/go/rpk/pkg/cli/cmd/cluster/config/reset.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package config
+
+import (
+	"path/filepath"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
+	"gopkg.in/yaml.v2"
+)
+
+func newResetCommand(fs afero.Fs) *cobra.Command {
+	var configCacheFile string
+	cmd := &cobra.Command{
+		Use:   "reset [PROPERTY...]",
+		Short: "Forcibly clear a cluster configuration property on this node.",
+		Long: `Forcibly clear a cluster configuration property on this node.
+
+This command is not for general changes to cluster configuration: use this only
+when redpanda will not start due to a configuration issue.
+
+This command erases a named property from the local node's internal 
+cache, so that on next startup redpanda will treat the setting as if it 
+was set to the default.
+`,
+		Run: func(cmd *cobra.Command, propertyNames []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			dataDir := cfg.Redpanda.Directory
+
+			// Same filename as in redpanda config_manager.cc
+			if configCacheFile == "" {
+				const cacheFileName = "config_cache.yaml"
+				configCacheFile = filepath.Join(dataDir, cacheFileName)
+			}
+
+			// Read YAML
+			f, err := afero.ReadFile(fs, configCacheFile)
+			out.MaybeDie(err, "Couldn't read %q", configCacheFile)
+
+			// Decode YAML
+			content := make(map[string]interface{})
+			err = yaml.Unmarshal(f, content)
+			out.MaybeDie(err, "Couldn't parse %q: %v", configCacheFile, err)
+
+			// Snip out the value we are resetting
+			for _, pn := range propertyNames {
+				delete(content, pn)
+			}
+
+			// Encode output
+			out_bytes, err := yaml.Marshal(content)
+			out.MaybeDie(err, "Serialization error: %v", configCacheFile, err)
+
+			// Write back output
+			err = afero.WriteFile(fs, configCacheFile, out_bytes, 0755)
```

Comment:
Is it bad if this write fails? Should we write a temp file and rename? Should this have a backup first that is then removed?

Reply:
This file isn't precious, fortunately -- it's just a cache of what's stored in raft0.  When we do a 'reset' we're intentionally fiddling with that cache to force the node to ignore one of the true raft0 values.  If something goes totally wrong writing this file, it can be deleted (or they can grab a copy off another node, as it's a common view of config).

In this kind of situation I'd usually expect the user to be acting at the behest of Support, so I'm inclined to keep the actual code as simple as possible rather than trying to handle all that automatically.